File: SigmaBooking.Backend/SigmaBooking.Domain/Services/TableService.cs
using SigmaBooking.Core.IServices;
using SigmaBooking.Core.Models;
using SigmaBooking.Domain.IRepositories;

namespace SigmaBooking.Domain.Services;

public class TableService : ITableService
{
    private readonly ITableRepository _tableRepository;

    public TableService(ITableRepository repository)
    {
        _tableRepository = repository;
    }
    public Table CreateTable(Table table)
    {
        return _tableRepository.CreateTable(table);
    }

    public List<Table> GetAllTables()
    {
        return _tableRepository.GetAllTables();
    }

    public Table UpdateTable(Table table)
    {
        return _tableRepository.UpdateTable(table);
    }

    public List<Table> UpdateTables(List<Table> tables)
    {
        return _tableRepository.UpdateTables(tables);
    }

    public void DeleteTable(string id)
    {
        _tableRepository.DeleteTable(id);
    }

    public Table GetTableById(string id)
    {
        return _tableRepository.GetTableById(id);
    }
}

File: SigmaBooking.Frontend/src/services/table.service.ts
import http from "./http.client";
import {Table} from "../models/Table";

export class TableService {
    async createTable(
        isStatic: boolean,
        x: number,
        y: number,
        w: number,
        h: number,
        i: string,
    ): Promise<Table[]> {
        const res = await http.post<Table[]>("/Tables", {
            static: isStatic,
            x: x,
            y: y,
            w: w,
            h: h,
            i: i
        });
        return res.data;
    }

    async getAllTables(): Promise<Table[]> {
        const res = await http.get("/Tables");
        return res.data.tables
    }

    getAll() {
        return http.get("/Tables")
    }
}
